mlx/backend/common/utils.cpp

```diff
@@ -11,6 +13,17 @@ std::string get_primitive_string(Primitive* primitive) {
   return op_t.str();
 }
 
+std::filesystem::path current_binary_dir() {
```

Comment:
There was a comment about this needing to stay inline. I think the purpose of that is to make sure upstream dependencies of MLX get the right binary path. Could you make it inline in the header?

Reply:
Maybe a better change would be to have a wrapper around the inline version which sets the static variable so it gets properly cached. But the one used by `register_library` in metal device.h should be inline in a header.

Reply:
The inline version of `register_library` got removed in https://github.com/ml-explore/mlx/pull/2055 (not sure if it is intentional), so I think there is no longer needs making this inline? The only place using it is `device.cpp` so it only returns the dir containing `libmlx.so`.

Reply:
Hmm.. good point. `get_library` can be called by third-party extensions. If it is called without a path then it will attempt to look for a metal library where the current binary is located. Presumably that will be the `libmlx` now.. which I would have thought is a bug 🤔

Reply:
Either way this PR doesn't change that behavior so let's move forward with it.